util.py

```diff
@@ -56,6 +57,40 @@ def __init__(self, creator, path_template, hidden_paths, cached=False):
         )
 
 
+class VariableString:
+    """
+    Enables to define strings containing Paths or Variables for e.g. ReturnnConfig or PipelineJob:
+    config[...] = VariableString("--some-parameter %s --another-parameter %i",
+                                 [sisyphus_path, sisyphus_variable])
```

Comment:
Why would this ever be needed?
You always should use sth like `["--some-parameter", sisyphus_path, "--another-parameter", sisyphus_variable]` instead, and that would not have this problem, right?

Reply:
This is not about a subprocess calls, but for any scenario where you want/need to pass a string to a job, but a part of that string depends on a sisyphus output. This should behave just like a tk.Variable/tk.Path.

Reply:
But that's what I mean, or asked about. I cannot imagine any case where this should be needed.
Or rather: If there is any such case, I think this is bad, and should be cleaned up anyway.